src/crimson/os/seastore/onode_manager/staged-fltree/stages/key_layout.h

```diff
@@ -545,11 +563,19 @@ class key_view_t {
   }
   std::string_view oid() const {
     // TODO(cross-node string dedup)
-    return ns_oid_view().oid.to_string_view();
+    return ns_oid_view().oid_or_lockey.to_string_view();
   }
   string_view_masked_t oid_masked() const {
     // TODO(cross-node string dedup)
-    return string_view_masked_t{ns_oid_view().oid};
+    return string_view_masked_t{ns_oid_view().oid_or_lockey};
+  }
+  std::string_view key() const {
+    // TODO(cross-node string dedup)
+    return oid();
+  }
+  string_view_masked_t key_masked() const {
+    // TODO(cross-node string dedup)
+    return oid_masked();
```

Comment:
It doesn't seem that we need both key() and oid() here, and the same to some other places around the onode index changes.

IIUC with this PR, the `oid` field of the onode tree index is replaced by the concept of `effective_key` of `hobject_t`, and there is no need to distingush key and oid from onode indexing perspective.

And if that can be aligned, how about splitting out the renaming efforts (oid -> effective_key) as a separate commit?

Reply:
But in the onode deltas, I think we need to remenber both oid and key, which is why I kept the `oid()` method

Reply:
> But in the onode deltas, I think we need to remenber both oid and key

Can you elaborate more about why need to remember both?